src/http_client/client.rs

```diff
@@ -0,0 +1,162 @@
+use crate::prelude::*;
+use serde_json::Value;
+
+/// A `HttpClient` needs a "network antenna" to be able to execute the
+/// network requests - which is a trait that clients implement on the FFI side (iOS/Android).
+pub struct HttpClient {
+    /// An object implementing the `NetworkAntenna` traits, which iOS/Android
+    /// clients pass into the constructor of this GatewayClient, so that it can
+    /// execute network requests.
+    pub network_antenna: Arc<dyn NetworkAntenna>,
+}
+
+impl HttpClient {
+    pub fn new(network_antenna: Arc<dyn NetworkAntenna>) -> Self {
+        Self { network_antenna }
+    }
+}
+
+impl HttpClient {
+    pub async fn execute_network_request(
+        &self,
+        request: NetworkRequest,
+    ) -> Result<BagOfBytes, CommonError> {
+        let response = self
+            .network_antenna
+            .execute_network_request(request)
+            .await?;
+
+        // Check for valid status code
+        if !(200..=299).contains(&response.status_code) {
+            return Err(CommonError::NetworkResponseBadCode);
+        }
+
```

Comment:
Note: previously of the body was empty, it did throw an error, but it is completely valid to get back an empty response for some requests. The validation moves to `model_from_response` where if a model is expected as response the decoding will fail for empty response.